apis/r/R/SOMASparseNDArray.R

```diff
@@ -174,57 +174,108 @@ SOMASparseNDArray <- R6::R6Class(
     #' @param coords Optional `list` of integer vectors, one for each dimension, with a
     #' length equal to the number of values to read. If `NULL`, all values are
     #' read. List elements can be named when specifying a subset of dimensions.
-    #' @template param-result-order
     #' @param repr Optional one-character code for sparse matrix representation type
     #' @param iterated Option boolean indicated whether data is read in call (when
     #' `FALSE`, the default value) or in several iterated steps.
     #' @param log_level Optional logging level with default value of `"warn"`.
-    #' @return A `matrix`-like object accessed using zero-based indexes. It supports
-    #'         only basic access operations with zero-based indexes as well as `dim()`,
-    #'         `nrow()`, and `ncol()`. Use `as.one.based()` to get a fully-featured
-    #'         sparse matrix object supporting more advanced operations (with one-based
-    #'         indexing).
-    read_sparse_matrix_zero_based = function(
+    #' @return A sparse matrix whose exact representation
+    #' is determined by \code{repr}.
+    read_sparse_matrix = function(
       coords = NULL,
       result_order = "auto",
       repr = c("C", "T", "R"),
       iterated = FALSE,
       log_level = "warn"
     ) {
       private$check_open_for_read()
-
+      repr <- repr[1L]
```

Comment:
I don't think you need that given the next line.  `match.arg()` will not let you pass a vector through.

Reply:
I've had `match.arg()` fail when passing default vectors, eg

```R
f <- function(repr = c("C", "T", "R")) {
  match.arg(repr)
}

g <- function(repr = c("C", "T", "R")) {
  f(x)
}

g()
```

but can't seem to reproduce it; I'll remove the line

Reply:
g() is 'outside the spec' methinks.  match.arg() does some black art _with the signature of its enclosing function_.  You break that in the example, you get to keep the pieces.

At least that's how it reads to me.

Reply:
Department of _Couldn't be quicker and dirtier if we tried_ calling on line 2:

```r
> f <- function(repr = c("C", "T", "R")) {
  match.arg(repr)
}
> f(c("T", "R", "R"))
Error in match.arg(repr) : 'arg' must be of length 1
> 
```

You had the right idea there but is already baked in.  And by default more explicit (--> actual error).